packages/core-sdk/src/resources/royalty.ts

```diff
@@ -48,8 +54,9 @@ export class RoyaltyClient {
   private readonly rpcClient: PublicClient;
   private readonly wallet: SimpleWalletClient;
   private readonly walletAddress: Address;
+  private readonly chainId: ChainIds;
 
-  constructor(rpcClient: PublicClient, wallet: SimpleWalletClient) {
+  constructor(rpcClient: PublicClient, wallet: SimpleWalletClient, chainId: ChainIds) {
```

Comment:
Shouldn't keep the singular (or plural) on both? I mean, either `chainId: ChainId` or `chainIds: ChainIds` ?

Reply:
yep, i added a backlog task to clean up our chain id usage throughout the code. I think the current method has alot of issues, including your point here. Won't be addressed in this PR.